tests/generation/test_utils.py

```diff
@@ -927,32 +928,44 @@ def test_prompt_lookup_decoding_stops_at_eos(self):
         self.assertTrue(output_prompt_lookup.shape[-1] == 10)
 
     @pytest.mark.generate
-    def test_left_padding_compatibility(self):
-        # NOTE: left-padding results in small numerical differences. This is expected.
-        # See https://github.com/huggingface/transformers/issues/25420#issuecomment-1775317535
+    def test_left_padding_compatibility(
```

Comment:
TL;DR
- better documented
- support of custom inputs (padded and unpadded)
- also prepare padded `token_type_ids`
- forward other model kwargs